### ISSUE TITLE
multi_asset_sensor fails with DagsterInvariantViolationError (Cannot fetch more than 10000 event records)

### ISSUE THREAD

**Issue by hota911** [dagster-io/dagster#23224]
### Dagster version

dagster, version 1.7.14

### What's the issue?

Our multi_asset_sensor start failing DagsterInvariantViolationError

```
dagster._core.errors.DagsterInvariantViolationError: Cannot fetch more than 10000 event records at a time. Requested 11775.
  File "/usr/local/lib/python3.11/site-packages/dagster/_grpc/impl.py", line 401, in get_external_sensor_execution
    return sensor_def.evaluate_tick(sensor_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/definitions/sensor_definition.py", line 807, in evaluate_tick
    result = self._evaluation_fn(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/definitions/sensor_definition.py", line 1153, in _wrapped_fn
    result.append(next(raw_evaluation_result))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/definitions/multi_asset_sensor_definition.py", line 1233, in _fn
    multi_asset_sensor_context.update_cursor_after_evaluation()
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/definitions/multi_asset_sensor_definition.py", line 390, in update_cursor_after_evaluation
    new_cursor = self._cursor_advance_state_mutation.get_cursor_with_advances(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/definitions/multi_asset_sensor_definition.py", line 861, in get_cursor_with_advances
    {
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/definitions/multi_asset_sensor_definition.py", line 862, in <dictcomp>
    str(asset_key): self.get_asset_cursor_with_advances(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/definitions/multi_asset_sensor_definition.py", line 900, in get_asset_cursor_with_advances
    context.instance.fetch_materializations(
  File "/usr/local/lib/python3.11/site-packages/dagster/_utils/__init__.py", line 684, in inner
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/instance/__init__.py", line 2043, in fetch_materializations
    return self._event_storage.fetch_materializations(records_filter, limit, cursor, ascending)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/storage/event_log/sql_event_log.py", line 1049, in fetch_materializations
    enforce_max_records_limit(limit)
  File "/usr/local/lib/python3.11/site-packages/dagster/_core/storage/event_log/sql_event_log.py", line 130, in enforce_max_records_limit
    raise DagsterInvariantViolationError(
```
![image](https://github.com/user-attachments/assets/3c038e77-6392-44e3-bbc6-2376ea605d5a)

It seems to fail fetching the monitored assets. However, monitored assets only have less than 100 materialization so not sure why it fails.

### What did you expect to happen?

Sensor works as expected.

### How to reproduce?

not sure 

### Deployment type

Other Docker-based deployment

### Deployment details

We deploy dagster on our k8s cluster

### Additional information

n/a

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by gibsondan** [comment]
Hi @hota911 - thanks for reporting this. We believe this was an issue introduced in the 1.7.14 release and we are working on a fix. In the short term I believe returning your code to 1.7.13 would make this go away while we sort this out.

**Comment by gibsondan** [comment]
Is it possible to share the name of the Dagster+ organization where you are seeing this?

**Comment by gibsondan** [comment]
Oh, sorry - just realized this an open source dagster deployment, ignore the request about an organization. Downgrading to 1.7.13 would still fix this, and also setting the MAX_LIMIT_GET_EVENT_RECORDS env var on your user code deployment to a very high number from the default of 10000 should also temporarily fix this.

**Comment by gibsondan** [comment]
We'll have a fix for this going out in the 1.7.15 release planned for later today.

**Comment by hota911** [comment]
@gibsondan Thank you for your quick reply. It's good to know fix will released soon.

JFYI, we workaround the issue by using advance_all_cursors instead of advance_cursor. It works fine.

It's not directly related to this issue, after having this sensor issue, we decided to implement a sensor to monitor sensors failures. 

- It would be great if there is an offitial way to do so. 
- To implement it ourselves, we need to workaround https://github.com/dagster-io/dagster/issues/17220 . It would be nice if you can fix it.

Thank you!
